mlx_graphs/transforms/base_transform.py

```diff
@@ -0,0 +1,21 @@
+import copy
+from abc import ABC
+from typing import Any
+
+
+class BaseTransform(ABC):
+    r"""An abstract class for creating transformations
+
+    Transforms are a way to modify and customize
+    ``GraphData`` by implicitly passing them as arguments
+    in ``Dataset``
+    """
+
+    def __call__(self, data: Any) -> Any:
+        return self.process(copy.copy(data))
+
+    def process(self, data: Any) -> Any:
```

Comment:
may be worth calling this transform and make it an abstract method so that its implementation is always required (like we did for process in the `Dataset` class)